### PR TITLE
KvikIO shared library

### PR DESCRIPTION
**Pull request by madsbk**
Update cudf  to use the new KvikIO shared library: https://github.com/rapidsai/kvikio/pull/527

#### Tasks
- [x] Wait for the [KvikIO shared library PR](https://github.com/rapidsai/kvikio/pull/527) to be merged.
- [x] Revert the use of the [KvikIO shared library](https://github.com/rapidsai/kvikio/pull/527) in CI: https://github.com/rapidsai/cudf/commit/2d8eeafe4959357a17f6ad488811837e0a07ba65.

### CODE REVIEW

File: cpp/cmake/thirdparty/get_kvikio.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@
 function(find_and_configure_kvikio VERSION)
 
   rapids_cpm_find(
-    KvikIO ${VERSION}
+    kvikio ${VERSION}
```
Comment (madsbk) [inline cpp/cmake/thirdparty/get_kvikio.cmake:19]: From https://github.com/rapidsai/cudf/pull/17243

**Comment by vyasr** [comment]
This looks much better. Now we're just hitting the cuda-python issues.

File: python/libcudf/CMakeLists.txt [hunk 2]
```diff
@@ -58,3 +61,12 @@ if(USE_NVCOMP_RUNTIME_WHEEL)
     APPEND
   )
 endif()
+
+if(USE_LIBKVIKIO_RUNTIME_WHEEL)
+  set(rpaths "$ORIGIN/../../libkvikio/lib64")
```
Comment (vyasr) [inline python/libcudf/CMakeLists.txt:66]: Instead of using RPATH we should call the load function. I'll update that locally and push.

**Review by vyasr — APPROVED** [review]

**Review by jameslamb — APPROVED** [review]

**Pull request merged**